### DISCUSSION TITLE
Compatibility with CosmosDB

### DISCUSSION THREAD

**Discussion by rudivanhierden** [apostrophecms/apostrophe#3880] [Q&A] [answered]
I could only find this issue regarding CosmosDB compatibility from 2018: https://github.com/apostrophecms/apostrophe/issues/1713 where @boutell said ApostropheCMS was made for MongoDB, and compatibility was not guaranteed.

Has there been any change in this? Are Apostrophe & CosmosDB still not 100% compatible?

**Comment by boutell** [comment, marked as answer]
We have pursued several experiments with CosmosDB over the years, often with the support of our friends at Michelin.

What we found is that while CosmosDB is much more compatible in the past, it is still not 100% compatible with the MongoDB API. Also, we found that we were charged an exorbitant amount for databases that were left in place even though we executed the MongoDB API calls to remove them. The Azure team did help us by waiving those charges on a one-time basis, and we are grateful for that, but the experience put a damper on our enthusiasm for CosmosDB.

Support for CosmosDB could come back to the front burner at some point if an enterprise customer wishes to support that work and provide resources to ensure bills for testing with CosmosDB are not a problem.

**Reply by rudivanhierden** [reply]
Thanks a lot for your quick and thorough answer! We'll probably stay with MongoDB or move to MongoDB Atlas — for now at least.
